controllers/k8ssandra/datacenters.go

```diff
@@ -308,7 +308,7 @@ func (r *K8ssandraClusterReconciler) reconcileDcRebuild(
 	task := &cassctlapi.CassandraTask{}
 
 	if err := remoteClient.Get(ctx, taskKey, task); err == nil {
-		if taskFinished(task) {
+		if taskFinished(task, int(dc.Spec.Size)) {
```

Comment:
Will we end up in endless loop if I change the size before the rebuild has finished (but has started) ? Since we reconcile the size parameter before the rebuild checks.

Reply:
Indeed we would. If we add `dc2` with a size of 3 and then decide to change the size to 4 (in the K8ssandraCluster spec) after the rebuild job has started, we would get into an endless requeue loop. 

I'll store the number of nodes to be rebuilt in a label on the CassandraTask and then I can just check the value of that label.